examples/chat_completion/chat_completion_functions_rest.py

```diff
@@ -0,0 +1,160 @@
+    data = {
+        "messages": messages,
+        "functions": functions,
+        "max_tokens": 20,
+        "temperature": 0.2,
+        "model": "gpt-3.5-turbo-0613",
+        "n": 1,
+    }
+
+    # The log_request function returns two other function for later logging
+    # the response or the exception. When we later do that, the logger will
+    # actually calculate all the relevant metrics and will send them to
+    # Mona.
+    response_logger, exception_logger = mona_logger.log_request(
+        data, additional_data={"customer_id": "A531251"}
+    )
+
+    try:
+        # Send the request to the API
+        response = requests.post(url, headers=headers, json=data)
+
+        # Check for HTTP errors
+        response.raise_for_status()
+
+        # Log response to Mona
+        response_logger(response.json())
+        print(response.json()["choices"][0]["message"])
+
+    except Exception:
+        # Log exception to Mona
+        exception_logger()
+
+    response_message = response.json()["choices"][0]["message"]
+
+    if response_message.get("function_call"):
+        available_functions = {
+            "get_current_weather": get_current_weather,
+        }
+        function_name = response_message["function_call"]["name"]
+        fuction_to_call = available_functions[function_name]
+        function_args = json.loads(
+            response_message["function_call"]["arguments"]
+        )
+        function_response = fuction_to_call(
+            location=function_args.get("location"),
+            unit=function_args.get("unit"),
+        )
+
+        messages.append(response_message)
+        messages.append(
+            {
+                "role": "function",
+                "name": function_name,
+                "content": function_response,
+            }
+        )
+
+        data = {
+            "messages": messages,
+            "functions": functions,
+            "max_tokens": 20,
+            "temperature": 0.2,
+            "model": "gpt-3.5-turbo-0613",
+            "n": 1,
+        }
+
+        # The log_request function returns two other function for later logging
+        # the response or the exception. When we later do that, the logger will
+        # actually calculate all the relevant metrics and will send them to
+        # Mona.
+        response_logger, exception_logger = mona_logger.log_request(
+            data, additional_data={"customer_id": "A531251"}
+        )
+
+        try:
+            # Send the request to the API
+            second_response = requests.post(url, headers=headers, json=data)
+
+            # Check for HTTP errors
+            second_response.raise_for_status()
+
+            # Log response to Mona
+            response_logger(second_response.json())
+            print(second_response.json()["choices"][0]["message"])
+            return second_response.json()
+
+        except Exception:
+            # Log exception to Mona
+            exception_logger()
```

Comment:
Lines 69-99 and 126-157 are very much the same - can be extracted to another function.